src/model/entities/test/rundown.spec.ts

```diff
@@ -2805,12 +2804,48 @@ describe(Rundown.name, () => {
           },
           infinitePieces: new Map(),
         },
+        segments: [
+          activeSegment,
+          nextSegment,
+        ],
       } as RundownInterface)
 
       testee.setNext(activeSegment.id, otherPartInActiveSegment.id)
 
       verify(mockedNextPart.reset()).once()
     })
+
+    describe('when next part is on air', () => {
+      it('throws an active part exception', () => {
+        const activePart: Part = EntityMockFactory.createPart({ id: 'active-part-id', isOnAir: true })
+        const mockedActiveSegment: Segment = EntityMockFactory.createSegmentMock({ id: 'active-segment-id', parts: [activePart] })
+        when(mockedActiveSegment.findPart(activePart.id)).thenReturn(activePart)
```

Comment:
Instead of mocking you could also use the `EntityTestFactory`.

Reply:
Updated to using the EntityTestFactory.